akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -50,7 +50,7 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
       if (totalDemand > 0) self ! Continue
     }
 
-  def readAndEmit() = try {
+  def readAndEmit(): Unit = try if (totalDemand > 0) {
```

Comment:
What would be the difference? (assuming that the comparison cannot possibly result in throwing an exception)